test/1_stdlib/Collection.swift

```diff
@@ -202,5 +202,41 @@ func testIsEmptyFirstLast() {
 }
 testIsEmptyFirstLast()
 
+/// A `CollectionType` that vends just the default implementations for
+/// `CollectionType` methods.
+struct CollectionOnly<T: CollectionType> : CollectionType {
+  var base: T
+
+  var startIndex: T.Index {
+    return base.startIndex
+  }
+
+  var endIndex: T.Index {
+    return base.endIndex
+  }
+
+  func generate() -> T.Generator {
+    return base.generate()
+  }
+
+  subscript(position: T.Index) -> T.Generator.Element {
+    return base[position]
+  }
+}
+
+func testFirstPerformance() {
+  // accessing `first` should not perform duplicate work on lazy collections
+  let col_ = (0..<10).lazy.filter({ print("filter: \($0)"); return (2..<8).contains($0) })
```

Comment:
@gribozavr Doesn't `StdlibUnittest` rely on existing with a non-zero error code to tell `lit.py` that it failed? Running the file and piping the results into `FileCheck` would suppress that error code. I suppose I could run it and pipe the output to a file, and then separately run `FileCheck` on that output, but I don't see any examples of doing this in the stdlib tests right now. Do we actually have any examples of a stdlib test that uses both FileCheck and `StdlibUnittest` tests?

@dabrahams I figured that the FileCheck tests were the old way of doing things, I'm just not sure how to mix the two in a way that works reliably.